init/modules/EESSI/2023.06.lua

```diff
@@ -13,12 +13,24 @@ conflict("EESSI")
 -- this is a version-agnostic module file, works for EESSI/2023.06, EESSI/2025.06, etc.
 local eessi_version = myModuleVersion()
 local eessi_repo = "/cvmfs/software.eessi.io"
+local eessi_archdetect_prefix = pathJoin(eessi_repo, "versions", eessi_version, "init")
+-- for RISC-V clients we need to do some overrides, as things are stored in different CVMFS repositories
 if (subprocess("uname -m"):gsub("\n$","") == "riscv64") then
-    eessi_version = os.getenv("EESSI_VERSION_OVERRIDE") or "20240402"
-    eessi_repo = "/cvmfs/riscv.eessi.io"
-    LmodMessage("RISC-V architecture detected, but there is no RISC-V support yet in the production repository.\n" ..
-                "Automatically switching to version " .. eessi_version .. " of the RISC-V development repository " .. eessi_repo .. ".\n" ..
-                "For more details about this repository, see https://www.eessi.io/docs/repositories/riscv.eessi.io/.")
+    if (eessi_version == "2023.06") then
+        eessi_version = os.getenv("EESSI_VERSION_OVERRIDE") or "20240402"
+        eessi_repo = "/cvmfs/riscv.eessi.io"
+        LmodMessage("RISC-V architecture detected, but there is no RISC-V support yet in the production repository.\n" ..
+                    "Automatically switching to version " .. eessi_version .. " of the RISC-V development repository " .. eessi_repo .. ".\n" ..
+                    "For more details about this repository, see https://www.eessi.io/docs/repositories/riscv.eessi.io/.")
+    elseif (eessi_version == "2025.06") then
+        eessi_repo = "/cvmfs/dev.eessi.io/riscv"
+        LmodMessage("This EESSI production version only provides a RISC-V compatibility layer,\n" ..
+                    "software installations are provided by the EESSI development repository at " .. eessi_repo .. ".\n")
+        if not isDir(eessi_repo) then
+            LmodError("The EESSI development repository dev.eessi.io is not mounted on your system.\n" ..
+                      "This is required for RISC-V systems.")
+        end
+    end
 end
 local eessi_prefix = pathJoin(eessi_repo, "versions", eessi_version)
```

Comment:
For consistency, this should be moved up and used to define `eessi_archdetect_prefix`?

Reply:
Done in https://github.com/EESSI/software-layer-scripts/pull/80/commits/001cea1b68c4a815f8dc7c17899f80e22d8b50e7. It does mean that we have to override that variable as well for the different RISC-V versions, but I've added that too in the same commit.